app/controllers/katello/api/v2/hosts_bulk_actions_controller.rb

```diff
@@ -247,6 +247,23 @@ def resolve_traces
       render json: result
     end
 
+    api :PUT, "/hosts/bulk/system_purpose", N_("Assign system purpose attributes on one or more hosts")
+    param_group :bulk_params
+    param :service_level, String, :desc => N_("Service level of host")
+    param :purpose_role, String, :desc => N_("Role of host")
+    param :purpose_usage, String, :desc => N_("Usage of host")
+    param :purpose_addons, Array, :desc => N_("Sets the system add-ons")
+    def system_purpose
```

Comment:
How does this API behave when I don't provide one of the params ? What gets set on the host, and what happens to the existing value?

IMO the behavior should be something like:
- not passing the param at all (nil value) should not modify the given syspurpose value
- passing in an empty string (or empty array in case of addons) should unset the addons

Reply:
Would the correct place to address this be here or in the task? The task seems like the appropriate place for it.

Reply:
Agree on the task - passing the params from the controller without any checking or manipulation here is ideal - can centralize that in the task

Reply:
Updated